src/plugins/sailfishoswizards/desktopwizardpages.cpp

```diff
@@ -0,0 +1,602 @@
+                                                      && QPixmap(field("icon108Path").toString()).width() == 108,
+                                                      m_pageUi.icon108Path, m_pageUi.errorLabel, tr("This file isn't 108x108 size image"))
+                && DesktopLocationPage::validateInput(!field("icon86Path").toString().isEmpty(),
+                                                      m_pageUi.icon86Path,
+                                                      m_pageUi.errorLabel, tr("The path of 86x86 image file should not be empty"))
+                && DesktopLocationPage::validateInput(QFile(field("icon86Path").toString()).exists(),
+                                                      m_pageUi.icon86Path,
+                                                      m_pageUi.errorLabel, tr("File of 86x86 icon with this path isn't exists"))
+                && DesktopLocationPage::validateInput(db.mimeTypeForFile(
+                                                          field("icon86Path").toString()).name().contains(
+                                                          QDir::toNativeSeparators("image/")),
+                                                      m_pageUi.icon86Path,
+                                                      m_pageUi.errorLabel, tr("File of 86x86 icon with this path isn't image"))
+                && DesktopLocationPage::validateInput(QPixmap(field("icon86Path").toString()).height() == 86
+                                                      && QPixmap(field("icon86Path").toString()).width() == 86,
+                                                      m_pageUi.icon86Path, m_pageUi.errorLabel, tr("This file isn't 86x86 size image"));
+}
+
+/*!
+ * \brief Clean up page's input fields.
+ */
+void DesktopIconPage::cleanupPage()
+{
+    m_pageUi.iconPath->clear();
+    m_pageUi.icon172Path->clear();
+    m_pageUi.icon128Path->clear();
+    m_pageUi.icon108Path->clear();
+    m_pageUi.icon86Path->clear();
+    m_pageUi.sizeLabel->clear();
+    m_pageUi.iconLabel->clear();
+    m_pageUi.manualInputCheck->setChecked(false);
+}
+
+/*!
+ * \brief Сontrols the activation of icons input fields.
+ */
+void DesktopIconPage::updateInputFields()
+{
+    if (m_pageUi.manualInputCheck->isChecked()) {
+        m_pageUi.icon172Path->setEnabled(true);
+        m_pageUi.icon172Browse->setEnabled(true);
+
+        m_pageUi.icon128Path->setEnabled(true);
+        m_pageUi.icon128Browse->setEnabled(true);
+
+        m_pageUi.icon108Path->setEnabled(true);
+        m_pageUi.icon108Browse->setEnabled(true);
+
+        m_pageUi.icon86Path->setEnabled(true);
+        m_pageUi.icon86Browse->setEnabled(true);
+
+        m_pageUi.iconPath->setEnabled(false);
+        m_pageUi.iconBrowse->setEnabled(false);
+        m_pageUi.iconPath->clear();
+        m_pageUi.iconLabel->clear();
+        m_pageUi.sizeLabel->clear();
+    } else {
+        m_pageUi.icon172Path->setText("");
```

Comment:
Maybe use `clear()` like in the code above?

Reply:
Yes. Fixed for all